src/java/org/apache/cassandra/service/CassandraDaemon.java

```diff
@@ -531,7 +540,12 @@ public void start()
             logger.info(isx.getMessage());
             return;
         }
+        startClientTransports();
```

Comment:
The `validateTransportsCanStart()` before this line seems no longer needed? Because both `startNativeTransport()` and `startThriftServer()` does the same validation before starting the service. 
If removing the duplicated validation, the `start()` method barely contains anything other than `startClientTransports()`.

Reply:
if both native and thrift are disabled, then the check doesn't happen in the start method, so the above check would be the only place that calls it; if either are enabled, then you are right that it is duplicate.

Reply:
If the instance is not going to start both native and thrift, would the validation be useless? It is to `validateTransportsCanStart`.

Reply:
if you choose to not start either protocol then validation only causes a log, but we will reject starting based off validation (was setup in https://issues.apache.org/jira/browse/CASSANDRA-14525), so I think its kinda useless.  It also causes a edge case where my new log doesn't happen since it would return.

I will remove and rerun python dtests.